module_utils/certificate/providers/base.py

```diff
@@ -526,6 +528,29 @@ def certificate_key_path(self):
         """Path where key should be placed."""
         return self._get_store_location(key=True)
 
+    def _get_hooks_directory(self, dirname):
+        script_name = os.path.basename(self.module.params.get("name")) + ".sh"
+        provider_config_directory = self.module.params.get("provider_config_directory")
+        return os.path.join(provider_config_directory, dirname, script_name)
```

Comment:
do you need to do something like `mkdir -p $provider_config_directory/$dirname/$(dirname $script_name))`?

Reply:
I've done that in the tasks. Do you think it's better to perform it here?

Reply:
> I've done that in the tasks. Do you think it's better to perform it here?

no, tasks is fine - I just didn't see it here in this file

Reply:
ok - I see where you create `$provider_config_directory/pre-scripts` and `post-scripts` - but e.g. if you have `name: /path/to/mycert` - don't you need to create `$provider_config_directory/pre-scripts/path/to` in order to write `$provider_config_directory/pre-scripts/path/to/mycert.sh`?

Reply:
ah, I see - there is a `basename` there

Reply:
There is a potential conflict - if the user specifies `name: /path/A/mycert` and `name: /path/B/mycert` - this code will attempt to create two different `pre-scripts/mycert.sh` - you might need to use the `sha` hash in the filename e.g. `pre-scripts/mycert-$hashval.sh` or something like that

Reply:
well observed... it would be really nice if name was actually a single identifier but it's not...

Reply:
What about a method in the provider to define an ID for the CSR? Certmonger already have a request-id. This way we could suffix the script with that.

Reply:
not sure which way is better, but an alternative is to create `$provider_config_directory/pre-scripts/path/A/mycert.sh` and `$provider_config_directory/pre-scripts/path/B/mycert.sh`

Reply:
> What about a method in the provider to define an ID for the CSR? Certmonger already have a request-id. This way we could suffix the script with that.

ok - will you have the request-id at this point in the code?

Reply:
We do. It's in `self._certmonger_metadata`.

Reply:
ok

Reply:
There is actually no certmonger used in centos/rhel. I'll leave group/owner as root and mode as 0770.

Reply:
Amended.